__tests__/table_moveRow.js

```diff
@@ -0,0 +1,489 @@
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('2')), td({ colspan: 2 }, p('merged cell'))),
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+    });
+
+    it('should move columns merged at last line', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('2')), tdEmpty, tdEmpty),
+            row(td(p('3')), td({ colspan: 2 }, p('merged cell')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('2')), tdEmpty, tdEmpty),
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('3')), td({ colspan: 2 }, p('merged cell')))
+          )
+        )
+      );
+    });
+
+    it('should move and keep table headers', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(th({ colspan: 2 }, p('merged cell')), thEmpty),
+            row(tdEmpty, tdEmpty, tdEmpty),
+            row(tdEmpty, tdEmpty, tdEmpty)
+          )
+        )
+      );
+
+      const newTr = moveRow(0, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(thEmpty, thEmpty, thEmpty),
+            row(tdEmpty, tdEmpty, tdEmpty),
+            row(td({ colspan: 2 }, p('merged cell')), tdEmpty)
+          )
+        )
+      );
+    });
+  });
+
+  describe('on a table with merged rows', () => {
+    it('should move rows', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td(p('3'))),
+            row(td(p('4')), td(p('5')), td(p('6')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('4')), td(p('5')), td(p('6'))),
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td(p('3')))
+          )
+        )
+      );
+    });
+
+    it('should not move rows for multi rows merged', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('---merged-row----')),
+              td(p('0')),
+              td(p('1'))
+            ),
+            row(td(p('2')), td({ rowspan: 2 }, p('---merged-row----'))),
+            row(td(p('4')), td(p('5')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 2)(tr);
```

Comment:
it should return an original `tr`

Reply:
We are not returning the original tr, since we are using cloneTr